common/client/multi_node_test.go

```diff
@@ -0,0 +1,629 @@
+		nodeSelector.On("Select").Return(nil).Once()
+		nodeSelector.On("Name").Return("MockedNodeSelector").Once()
+		mn.nodeSelector = nodeSelector
+		node, err := mn.selectNode()
+		require.EqualError(t, err, ErroringNodeError.Error())
+		require.Nil(t, node)
+		tests.RequireLogMessage(t, observedLogs, "No live RPC nodes available")
+
+	})
+}
+
+func TestMultiNode_nLiveNodes(t *testing.T) {
+	t.Parallel()
+	type nodeParams struct {
+		BlockNumber     int64
+		TotalDifficulty *utils.Big
+		State           nodeState
+	}
+	testCases := []struct {
+		Name                    string
+		ExpectedNLiveNodes      int
+		ExpectedBlockNumber     int64
+		ExpectedTotalDifficulty *utils.Big
+		NodeParams              []nodeParams
+	}{
+		{
+			Name:                    "no nodes",
+			ExpectedTotalDifficulty: utils.NewBigI(0),
+		},
+		{
+			Name:                    "Best node is not healthy",
+			ExpectedTotalDifficulty: utils.NewBigI(10),
+			ExpectedBlockNumber:     20,
+			ExpectedNLiveNodes:      3,
+			NodeParams: []nodeParams{
+				{
+					State:           nodeStateOutOfSync,
+					BlockNumber:     1000,
+					TotalDifficulty: utils.NewBigI(2000),
+				},
+				{
+					State:           nodeStateAlive,
+					BlockNumber:     20,
+					TotalDifficulty: utils.NewBigI(9),
+				},
+				{
+					State:           nodeStateAlive,
+					BlockNumber:     19,
+					TotalDifficulty: utils.NewBigI(10),
+				},
+				{
+					State:           nodeStateAlive,
+					BlockNumber:     11,
+					TotalDifficulty: nil,
+				},
+			},
+		},
+	}
+
+	chainID := types.RandomID()
+	mn := newTestMultiNode(t, multiNodeOpts{
+		selectionMode: NodeSelectionModeRoundRobin,
+		chainID:       chainID,
+	})
+	for i := range testCases {
+		tc := testCases[i]
+		t.Run(tc.Name, func(t *testing.T) {
+			for _, params := range tc.NodeParams {
+				node := newMockNode[types.ID, types.Head[Hashable], multiNodeRPCClient](t)
+				node.On("StateAndLatest").Return(params.State, params.BlockNumber, params.TotalDifficulty)
+				mn.nodes = append(mn.nodes, node)
+			}
+
+			nNodes, blockNum, td := mn.nLiveNodes()
+			assert.Equal(t, tc.ExpectedNLiveNodes, nNodes)
+			assert.Equal(t, tc.ExpectedTotalDifficulty, td)
+			assert.Equal(t, tc.ExpectedBlockNumber, blockNum)
+		})
+	}
+}
+
+func TestMultiNode_BatchCallContextAll(t *testing.T) {
+	t.Parallel()
+	t.Run("Fails if failed to select active node", func(t *testing.T) {
+		chainID := types.RandomID()
+		mn := newTestMultiNode(t, multiNodeOpts{
+			selectionMode: NodeSelectionModeRoundRobin,
+			chainID:       chainID,
+		})
+		nodeSelector := newMockNodeSelector[types.ID, types.Head[Hashable], multiNodeRPCClient](t)
+		nodeSelector.On("Select").Return(nil).Once()
+		nodeSelector.On("Name").Return("MockedNodeSelector").Once()
+		mn.nodeSelector = nodeSelector
+		err := mn.BatchCallContextAll(tests.Context(t), nil)
+		require.EqualError(t, err, ErroringNodeError.Error())
+	})
+	t.Run("Returns response of active node", func(t *testing.T) {
```

Comment:
This message implies a happy path. Here the RPC returns an error. Perhaps a more fitting name would be:
```suggestion
	t.Run("Returns error if RPC call fails for active node", func(t *testing.T) {
```